src/main/java/org/folio/qm/service/impl/RecordCreationServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+package org.folio.qm.service.impl;
+
+import static java.util.Objects.requireNonNull;
+
+import static org.folio.qm.util.ChangeManagerPayloadUtils.getDefaultJobProfile;
+import static org.folio.qm.util.ChangeManagerPayloadUtils.getDefaultJodExecutionDto;
+import static org.folio.qm.util.ChangeManagerPayloadUtils.getRawRecordsBody;
+import static org.folio.qm.util.JsonUtils.objectToJsonString;
+import static org.folio.qm.util.MarcUtils.updateRecordTimestamp;
+import static org.folio.qm.util.StatusUtils.getStatusInProgress;
+import static org.folio.qm.util.StatusUtils.getStatusNew;
+
+import java.util.UUID;
+import java.util.function.Predicate;
+
+import lombok.RequiredArgsConstructor;
+import org.mapstruct.ap.internal.util.Strings;
+import org.springframework.stereotype.Component;
+
+import org.folio.qm.client.SRMChangeManagerClient;
+import org.folio.qm.config.properties.JobExecutionProfileProperties;
+import org.folio.qm.converter.MarcConverterFactory;
+import org.folio.qm.domain.dto.CreationStatus;
+import org.folio.qm.domain.dto.FieldItem;
+import org.folio.qm.domain.dto.MarcFormat;
+import org.folio.qm.domain.dto.QuickMarc;
+import org.folio.qm.mapper.CreationStatusMapper;
+import org.folio.qm.service.CreationStatusService;
+import org.folio.qm.service.RecordCreationService;
+import org.folio.rest.jaxrs.model.InitialRecord;
+import org.folio.spring.FolioExecutionContext;
+
+@Component
+@RequiredArgsConstructor
+public class RecordCreationServiceImpl implements RecordCreationService {
+
+  private final SRMChangeManagerClient srmClient;
+  private final CreationStatusService statusService;
+  private final JobExecutionProfileProperties jobExecutionProfileProperties;
+  private final FolioExecutionContext folioExecutionContext;
+  private final CreationStatusMapper statusMapper;
+  private final MarcConverterFactory marcConverterFactory;
+
+  @Override
+  public CreationStatus createRecord(QuickMarc quickMarc) {
+    final var userId = folioExecutionContext.getUserId().toString();
+
+    JobExecutionProfileProperties.ProfileOptions options = null;
+    if (MarcFormat.BIBLIOGRAPHIC == quickMarc.getMarcFormat()) {
```

Comment:
What happens if the record is in marcFormat: AUTHORITY ? Maybe we should add verification for AUTHORITY?